swap_meet/vendor.py

```diff
@@ -0,0 +1,106 @@
+class Vendor():
+    def __init__(self, inventory = False):
+        self.inventory = list() 
+        
+        if inventory:
+            self.inventory = inventory
+      
+    
+    def add(self, add_item):
+        self.inventory.append(add_item)
+        return add_item
+    
+    def remove(self, remove_item):
+        if remove_item in self.inventory:
+            self.inventory.remove(remove_item)
+            return remove_item
+        else:
+            return False
+    
+    #wave2
+    def get_by_category(self, category):
+        matching_items = []
+
+        for item in self.inventory:
+            if item.category == category:
+                matching_items.append(item)
+        return matching_items
+
+    #wave3
+    def swap_items(self, friend, my_item, their_item):
+        if my_item not in self.inventory or their_item not in friend.inventory: 
+            return False
+
+        self.remove(my_item)
+        friend.add(my_item) #add to vendor
+        friend.remove(their_item)
+        self.add(their_item)
+        return True
+#wave4
+    def swap_first_item(self, friend):
```

Comment:
👍 , Just wanted to point out that you can use `swap_item` in the `swap_first_item` function.